src/apps/testapps/testPolyfill.c

```diff
@@ -262,6 +231,35 @@ TEST(polyfillTransmeridian) {
     free(hexagonsTMH);
 }
 
+TEST(polyfillTransmeridianComplex) {
+    // This polygon is "complex" in that it has > 4 vertices - this
+    // tests for a bug that was taking the max and min longitude as
+    // the bounds for transmeridian polygons
+    GeoCoord verts[] = {{0.1, -M_PI + 0.00001},  {0.1, M_PI - 0.00001},
+                        {0.05, M_PI - 0.2},      {-0.1, M_PI - 0.00001},
+                        {-0.1, -M_PI + 0.00001}, {-0.05, -M_PI + 0.2}};
+
+    Geofence geofence;
+    geofence.verts = verts;
+    geofence.numVerts = 6;
+
+    GeoPolygon polygon;
+    polygon.geofence = geofence;
+    polygon.numHoles = 0;
+
+    int numHexagons = H3_EXPORT(maxPolyfillSize)(&polygon, 4);
+
+    H3Index* hexagons = calloc(numHexagons, sizeof(H3Index));
+    H3_EXPORT(polyfill)(&polygon, 4, hexagons);
+
+    int actualNumHexagons = countActualHexagons(hexagons, numHexagons);
+
+    t_assert(actualNumHexagons == 1204,
```

Comment:
If we keep `t_assert` as immediately exiting program, this is fine. Changing that to another behavior might result in a memory leak if we don't hit line 260.

Reply:
This would be true in a lot of places. I'm assuming the current behavior is fine in this test, otherwise we'd have the same problem all over. If `t_assert` doesn't exit the program, I'd assume it doesn't exit the function either, so we'd still hit L260.

Reply:
Ya I have a feeling that's true. OK for now, assertions will remain fatal.